programs/merkle-distributor/src/instructions/admin/new_distributor.rs

```diff
@@ -0,0 +1,225 @@
+use crate::error::ErrorCode::ArithmeticError;
+use crate::state::merkle_distributor::{ActivationType, ClaimType};
+use crate::{
+    error::ErrorCode,
+    state::merkle_distributor::{AirdropBonus, MerkleDistributor},
+};
+use anchor_lang::{account, context::Context, prelude::*, Accounts, Key, ToAccountInfo};
+use anchor_spl::token::{Mint, Token, TokenAccount};
+
+#[cfg(feature = "localnet")]
+const SECONDS_PER_DAY: i64 = 0;
+
+#[cfg(not(feature = "localnet"))]
+const SECONDS_PER_DAY: i64 = 24 * 3600; // 24 hours * 3600 seconds
+
+#[derive(AnchorSerialize, AnchorDeserialize, InitSpace)]
+pub struct NewDistributorParams {
+    pub version: u64,
+    pub root: [u8; 32],
+    pub total_claim: u64,
+    pub max_num_nodes: u64,
+    pub start_vesting_ts: i64,
+    pub end_vesting_ts: i64,
+    pub clawback_start_ts: i64,
+    pub activation_point: u64, // can be slot or timestamp
+    pub activation_type: u8,
+    pub closable: bool,
+    pub total_bonus: u64,
+    pub bonus_vesting_duration: u64,
+    pub claim_type: u8,
+    pub operator: Pubkey,
+    pub locker: Pubkey,
```

Comment:
Should we move `locker` into `NewDistributor` context so it will validate and make sure it's a valid locker account?
```
pub struct NewDistributor<'info> {
...
	pub locker: Option<Box<Account<'info, Locker>>>
}
```

Reply:
I think it is fine, admin must re-validate distributor after it is initialized